lesson10/js/main.js

```diff
@@ -0,0 +1,71 @@
+"use strict";
+const menuContainer = document.getElementById("js-menu_container");
+const menuLists = document.createElement("ul");
+const menuContents = [
+  {
+    to: "bookmark.html",
+    img: "images/1.png",
+    alt: "画像1",
+    text: "ブックマーク",
+  },
+  {
+    to: "message.html",
+    img: "images/2.png",
+    alt: "画像2",
+    text: "メッセージ",
+  },
+];
+
+const loading = document.getElementById("js-loading");
+
+const loadingImg = document.createElement("img");
+loadingImg.src = "images/loading-circle.gif";
+loadingImg.classList.add("el_loadingImg");
+
+function createMenuList(values) {
+  values.forEach((value) => {
+    const listItem = document.createElement("li");
+
+    const listImg = document.createElement("img");
+    listImg.src = value.img;
+    listImg.alt = value.alt;
+    listImg.width = 40;
+    listImg.height = 40;
+
+    const listAnchor = document.createElement("a");
+    listAnchor.href = value.to;
+    listAnchor.textContent = value.text;
+
+    const fragment = document.createDocumentFragment();
+    fragment.appendChild(listAnchor).appendChild(listImg);
+    listItem.appendChild(fragment);
+
+    menuContainer.appendChild(menuLists).appendChild(listItem);
+  });
+}
+
+const fetchMenuContents = new Promise((resolve, reject) => {
+  loading.appendChild(loadingImg);
+  const checkArray = Object.keys(menuContents);
+  setTimeout(() => {
+    if (checkArray.length > 0 || !checkArray == null) {
```

Comment:
ここは`checkArray.length > 0`で良いのではないかと思いました。
`Object.keys`は`与えられたオブジェクトのすべての列挙可能なプロパティを表す文字列の配列`が返ってきます。
もし中に何も入っていない配列が格納されていたとしても、`空配列`を返します。
であればその場合の`length`は0になると思います。
その時点で、右の`!checkArray == null`に評価が移ることはありません。
もしifをもっと厳格にしたい場合には`&&`の方が適切かなと思いました。

`!checkArray == null`とするのであれば、明示的に`null`ではないことを表しているので、
戻り値が`null`となる処理に書くのであれば良いと思いました。
例えば`getElementById`とかですね。
https://developer.mozilla.org/ja/docs/Web/API/Document/getElementById
> 指定された ID に一致する DOM 要素オブジェクトを記述した Element オブジェクト、または文書内に一致する要素がなければ null です。

Reply:
あと、できるだけ`==`よりも`===`を使うようにしたいです。
ここの辺は結構厳密にしていた方がのちのちのエラーに繋がりにくいです。
https://qiita.com/Nossa/items/358c467906c38cd04f91

加えて、`!checkArray == null`とするよりも`checkArray !== null`としたいです。

おそらく、やりたかったこととしては
「checkArrayがnullではなかったら」
だと思うのですが、`!checkArray == null`としてしまうと
「checkArrayがnullだったとき」
と意味が逆転してしまいます。
なので、if文の条件が
「配列の中身が0より大きい時、またはcheckArrayの結果がnullだったとき」
と、辻褄が合わなくなっている状態です。
そのため、`checkArray !== null`の方が適切ではないかと思います。
`checkArray !== null`であれば、
「checkArrayがnullでなければ」
という意味になります。

https://www.javadrive.jp/javascript/ope/index4.html#section2

Reply:
> もしifをもっと厳格にしたい場合には`&&`の方が適切かなと思いました。
すみません、こちら  ||  にされていた意図を間違えて捉えていました。
この一文は無視してください🙇‍♀️

Reply:
Thank you for review.

I noticed that an empty array was created by using Object.keys.
I simply changed it to length only.

How to write "!" I thought I could only write at the beginning.
So, how to write a comparison I learned a lot.

[d468ab4](https://github.com/nyattsu72/hanson-frontend-cource/pull/13/commits/d468ab47d814b342d22ce703741eab3a66ead858)


Question
-----------------------------------------------------------
I didn't reflect it in this code, but I have a question.

The commit code will fail if the array is 0.
However, if the array is 0 or more, but the data content is insufficient, it will be resolved.

If there is no image in the data, it will be displayed without an image.
I thought this was incorrect.

In this case, should I check with the resolve condition?
Or should I check the data when storing it?

I thought it was time to store the data, so I didn't reflect it in this commit code.

Reply:
Thanks for the question.
But I don"t understand your question. 😢

For example, what's the state of the `menuList`?

this code?
```javascript
const menuList = [
  {
    to: "bookmark.html",
    // img: "images/1.png",
    alt: "画像1",
    text: "ブックマーク"
  },
  {
    to: "message.html",
    img: "images/2.png",
    alt: "画像2",
    text: "メッセージ"
  }
]
```

Reply:
Thank you for answering the question.
I understood from the answer you received.

I didn't understand that it was the backend side.
You don't have to check the contents of menuList on the front end
I understand that it is confirmed when storing data.

Thank you very much for your answers!